routes/auth.js

```diff
@@ -114,7 +115,40 @@ router.get('/users/:id', (req, res) => {
   });
 });
 
-// forgot route
+// User profile edit route
+router.get('/users/:id/edit', middleware.checkUserOwnership, (req, res) => {
+  User.findById(req.params.id, (err, foundUser) => {
+    if (err || !(foundUser && foundUser._id)) {
+      req.flash('error', 'Something went wrong');
+      res.redirect('/');
+    } else {
+      res.render('users/profile-edit', {
+        user: foundUser,
+      });
+    }
+  });
+});
+
+// Update profile route
+router.put('/users/:id', middleware.checkUserOwnership, (req, res) => {
+  // Find and update the correct user
+  const user = req.body.user;
```

Comment:
I'd suggest extracting only the safe keys from the `req.body.user` object and use those to update the existing user. I don't think you would want to add some non-existing properties or change if the user is admin, etc.